vm/interpreter/src/vm.rs

```diff
@@ -99,16 +97,13 @@ where
         reward_calc: Arc<dyn RewardCalc>,
         lb_fn: Box<dyn Fn(ChainEpoch) -> Cid>,
         engine: Engine,
-        heights: Heights,
         chain_finality: i64,
     ) -> Result<Self, anyhow::Error>
     where
         R: Rand + Clone + 'static,
-        C: FnOnce(ChainEpoch, &StateTree<&DB>) -> Result<TokenAmount, anyhow::Error>,
+        C: FnOnce(ChainEpoch, &DB) -> Result<TokenAmount, anyhow::Error>,
     {
-        let state = StateTree::new_from_root(&store_arc, &root)?;
```

Comment:
@lemmih Looks like it's possible to totally remove this callback and pass `circ_supply` as fn arg

Reply:
Go for it.

Reply:
Done.